wire-ios-data-model/Tests/MLS/CommitSenderTests.swift

```diff
@@ -0,0 +1,282 @@
+            // It clears pending commit
+            XCTAssertEqual(mockClearPendingCommitInvocations.count, 1)
+            XCTAssertEqual(mockClearPendingCommitInvocations.first, groupID.bytes)
+        } else {
+            // It doesn't clear pending commit
+            XCTAssertEqual(mockClearPendingCommitInvocations.count, 0)
+        }
+    }
+
+    // MARK: - Send External Commit
+
+    func test_SendExternalCommitBundle_Success() async throws {
+        // Given
+        let event = ZMUpdateEvent()
+
+        // Mock action provider
+        mockActionsProvider.sendCommitBundleIn_MockMethod = { _, _ in
+            return [event]
+        }
+
+        // Mock core crypto
+        var mergePendingGroupInvocations = [[Byte]]()
+        mockCoreCrypto.mockMergePendingGroupFromExternalCommit = { groupID in
+            mergePendingGroupInvocations.append(groupID)
+        }
+
+        // When
+        let receivedEvents = try await sut.sendExternalCommitBundle(commitBundle, for: groupID)
+
+        // Then
+        // Send commit bundle was called on the action provider
+        XCTAssertEqual(mockActionsProvider.sendCommitBundleIn_Invocations.count, 1)
+        XCTAssertEqual(mockActionsProvider.sendCommitBundleIn_Invocations.first?.bundle, commitBundle.transportData())
+
+        // Commit accepted was called
+        XCTAssertEqual(mergePendingGroupInvocations.count, 1)
+        XCTAssertEqual(mergePendingGroupInvocations.first, groupID.bytes)
+
+        // Events were received
+        XCTAssertEqual(receivedEvents, [event])
+    }
+
+    func test_SendExternalCommitBundle_ThrowsWithRecoveryStrategy_Retry() async {
+        await assertSendExternalCommitBundleThrows(
+            withRecovery: .retry,
+            for: .mlsStaleMessage,
+            shouldClearPendingGroup: false
+        )
+    }
+
+    func test_SendExternalCommitBundle_ThrowsWithRecoveryStrategy_GiveUp() async {
+        await assertSendExternalCommitBundleThrows(
+            withRecovery: .giveUp,
+            for: .unknown(status: 400, label: "", message: ""),
+            shouldClearPendingGroup: true
+        )
+    }
+
+    private func assertSendExternalCommitBundleThrows(
+        withRecovery recovery: ExternalCommitError.Recovery,
+        for error: SendCommitBundleAction.Failure,
+        shouldClearPendingGroup: Bool,
+        file: StaticString = #file,
+        line: UInt = #line
+    ) async {
+        // Given
+        // Mock action provider throwing an error
+        mockActionsProvider.sendCommitBundleIn_MockMethod = { _, _ in
+            throw error
+        }
+
+        // Then
+        await assertItThrows(error: ExternalCommitError.failedToSendCommit(recovery: recovery)) {
+            // When
+            _ = try await sut.sendExternalCommitBundle(commitBundle, for: groupID)
+        }
+
+        if shouldClearPendingGroup {
+            // It clears pending commit
+            XCTAssertEqual(mockClearPendingGroupInvocations.count, 1)
```

Comment:
The `file` and `line` arguments are not used.

Reply:
good catch! fixed here https://github.com/wireapp/wire-ios/pull/787/commits/b68cde7421b9d6c96ddab4468e3e05b899778eaf